### PR TITLE
Add download widget shortcode, Udpate code to avoid cross origin issue

### PR DESCRIPTION
**Pull request by thrijith**
Fixe #979

### CODE REVIEW

**Comment by thrijith** [comment]
I've checked the `URL` signing algorithm [here](https://docs.laterpay.net/platform/intro/signing_urls/#signing-urls) 

For single post, the URL to be added as a part of the base message ( Refer to point 6 of above URL ) is like this.

> http://lsvip.test/2018/10/15/paid-content-title/?post_id=50&buy=true

And for shortcode, it is something like this

> http://lsvip.test/2018/10/15/paid-content-title/?post_id=50&buy=true&action=laterpay_get_premium_shortcode_link&ids%5B%5D=50&types%5B%5D=file&post_id=49

I am not sure how are `ids` and `types` which are array themselves treated right now to create base message. It would be helpful if I could know how is the base message created for URL given for shortcode.

In our case the message generated is

<details>
<summary>Click me to expand/collapse</summary>
GET&http%3A%2F%2Flsvip.test%2F2018%2F10%2F15%2Fpaid-content-title%2F&action%3Dlaterpay_get_premium_shortcode_link%26buy%3Dtrue%26ids%3D50%26lptoken%3Dt%257C6IExtdNrI%252B1195Ig9cNfSkJGQKLjfNpq2UO5QpScNbblwOaupvdr9eJ%252BUscvFDpgQ%252FsIoh3SznwkvtdWUfBSkA%253D%253D%257C1539950058%257C8522f106111f0bf86bc085221c9e21aa89ce58e5f8ebd405184b3022%26post_id%3D49%26ts%3D1539950058%26types%3Dfile
</details>

---

Which after signing with the secret gives a different hmac than the one sent from api.
So it is creating problems with access. Please correct me if I missed something.

**Comment by thrijith** [comment]
HMAC recieved from API `d2e1cf6a2077a9b3494c6a9f0263a6cd6e1dd52489b7b171361e4108`

---

HMAC generated from library `fc9a2e68824a68b8becae2766fc6479da4c758802cd78418840670d0`

Just for reference:

The signing method in client library 👉  [this](https://github.com/laterpay/laterpay-client-php/blob/develop/LaterPay/Client/Signing.php#L171).

**Comment by thrijith** [comment]
HMAC recieved from API `d2e1cf6a2077a9b3494c6a9f0263a6cd6e1dd52489b7b171361e4108`

* * *

HMAC generated from library `fc9a2e68824a68b8becae2766fc6479da4c758802cd78418840670d0`

Just for reference:

The signing method in client library :point_right: [this](https://github.com/laterpay/laterpay-client-php/blob/develop/LaterPay/Client/Signing.php#L171).

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 1]
```diff
@@ -36,6 +44,174 @@ public static function get_subscribed_events() {
...
+                    . '" data-post-id="' . $page_id
+                    . '" data-content-type="' . $content_type
+                    . '" data-page-url="' . $page_url
+                    . '">';
+        }
+        // create a premium box
+        $html .= '    <div class="lp_premium-file-box__details">';
+        $html .= '        <h3 class="lp_premium-file-box__title">' . $heading . '</h3>';
+        if ( ! empty( $description ) ) {
+            $html .= '    <p class="lp_premium-file-box__text">' . $description . '</p>';
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Frontend/Shortcode.php]:
```suggestion
            $html .= '<p class="lp_premium-file-box__text">' . $description . '</p>';
```

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 2]
```diff
@@ -36,6 +44,174 @@ public static function get_subscribed_events() {
...
+        }
+        $html .= '    </div>';
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Frontend/Shortcode.php]:
```suggestion
        $html .= '</div>';
```

File: laterpay/application/Controller/Frontend/Shortcode.php [hunk 3]
```diff
@@ -36,6 +44,174 @@ public static function get_subscribed_events() {
...
+            }
+        } else if ( in_array( $a['content_type'], $content_types, true ) ) {
+            $content_type = $a['content_type'];
+        } else {
+            $content_type = 'text';
+        }
+        // escape user input
+        $image_path     = esc_url( $a['teaser_image_path'] );
+        $heading        = esc_attr( $a['heading_text'] );
+        $description    = esc_attr( $a['description_text'] );
```
Comment (PatelUtkarsh) [inline laterpay/application/Controller/Frontend/Shortcode.php]: Should late escape on markup below?

**Pull request merged**